backend/dal/Services/Concrete/ProjectService.cs

```diff
@@ -0,0 +1,285 @@
+                .Include(p => p.Properties).ThenInclude(p => p.Parcel).ThenInclude(p => p.Fiscals)
+                .Include(p => p.Properties).ThenInclude(b => b.Building)
+                .Include(p => p.Properties).ThenInclude(b => b.Building).ThenInclude(b => b.Evaluations)
+                .Include(p => p.Properties).ThenInclude(b => b.Building).ThenInclude(b => b.Fiscals)
+                .AsNoTracking()
+                .FirstOrDefault(p => p.ProjectNumber == projectNumber &&
+                    (isAdmin || userAgencies.Contains(p.AgencyId))) ?? throw new KeyNotFoundException();
+
+            // Remove any sensitive properties from the results if the user is not allowed to view them.
+            if (!viewSensitive)
+            {
+                project?.Properties.RemoveAll(p => p.Parcel.IsSensitive);
+                project?.Properties.RemoveAll(p => p.Building.IsSensitive);
+            }
+            return project;
+        }
+
+        /// <summary>
+        /// Add the specified project to the datasource.
+        /// </summary>
+        /// <param name="project"></param>
+        /// <returns></returns>
+        public Project Add(Project project)
+        {
+            project.ThrowIfNull(nameof(project));
+            this.User.ThrowIfNotAuthorized(Permissions.PropertyAdd);
+
+            var agency = this.User.GetAgency(this.Context) ??
+                throw new NotAuthorizedException("User must belong to an agency before adding projects.");
+
+            project.AgencyId = agency.Id; // Always assign the current user's agency to the project.
+            project.Agency = agency;
+            project.StatusId = 0; // Always start a project as a Draft.
+            project.Status = this.Context.ProjectStatus.Find(project.StatusId);
+            project.TierLevel = this.Context.TierLevels.Find(project.TierLevelId);
+
+            // If the tasks haven't been specified, generate them.
+            if (!project.Tasks.Any())
+            {
+                var tasks = this.Context.Tasks.Where(t => t.TaskType == TaskTypes.DisposalProjectDocuments);
+                foreach (var task in tasks)
+                {
+                    project.Tasks.Add(new ProjectTask(project, task));
+                }
+            }
+
+            this.Context.Projects.Add(project);
+            this.Context.CommitTransaction();
+            return project;
+        }
+
+        /// <summary>
+        /// Update the specified project in the datasource.
+        /// </summary>
+        /// <param name="project"></param>
+        /// <exception type="KeyNotFoundException">Entity does not exist in the datasource.</exception>
+        /// <returns></returns>
+        public Project Update(Project project)
+        {
+            project.ThrowIfNotAllowedToEdit(nameof(project), this.User, new[] { Permissions.PropertyEdit, Permissions.AdminProperties });
```

Comment:
can users without sensitive permissions edit projects with sensitive properties?

Reply:
Hmmm, I suspect we shouldn't allow them in that case.

Reply:
That may be an issue if we update this to support removing properties. Users without the permission will get a filtered list of properties, and when submitting that back to the DB it'll look like the sensitive property was removed.

Reply:
I think the easiest is just not allow them to edit in this case.